app/Http/Controllers/DashboardController.php

```diff
@@ -29,36 +27,42 @@ public function __construct()
      */
     public function index()
     {
-        try
-        {
-            //Todo use the users preferred device
-            //Get the first device that doesn't have a null location_id
-            $device = Device::publicDashData()->where('location_id', '!=', 'null')->orderBy('id', 'ASC')->first();
-        }
-        catch(ModelNotFoundException $e)
-        {
-            //If no device is found then redirect the user to the home page and display a message about registering a device
-            return redirect('home')->with('no_devices', 'To access the dashboard page there must be at least one registered device assigned to a location.');
-        }
-        
-        //Get the devices location
-        $location = $device->location()->select('id', 'name', 'site_id')->firstOrFail();
-        //Get the devices site
-        $site = $location->site()->select('id', 'name')->firstOrFail();
+        //TODO use the users preferred device
+        $device_id = Device::firstOrFail()->id;
```

Comment:
You should get the whole device here not just the id. Then you can use the relationships to grab its location, etc...
$device=Device::firstOrFail();